src/main/java/com/amazon/opendistroforelasticsearch/security/resolver/IndexResolverReplacer.java

```diff
@@ -359,48 +324,35 @@ public Resolved resolveRequest(final Object request) {
          */
         private static final Set<String> All_SET = Collections.singleton("*");
         private static final long serialVersionUID = 1L;
-        public final static Resolved _LOCAL_ALL = new Resolved(All_SET, All_SET, All_SET, All_SET, Collections.emptySet(), Collections.emptySet());
+        public final static Resolved _LOCAL_ALL = new Resolved(All_SET, All_SET, All_SET, Collections.emptySet());
         private final Set<String> aliases;
-        private final Set<String> indices;
         private final Set<String> allIndices;
-        private final Set<String> types;
+        private final Set<String> types = ImmutableSet.of("*");
```

Comment:
nit: make `type` `static` and remove from `equals()` and `hashCode()`

Reply:
Please remove reference to `type` from `equals()` and `hashCode()`.  `getType()` should return `type` (no need to wrap).

Reply:
Done